### PR TITLE
tatl: init at 1.0

### PR DESCRIPTION
**Pull request by mgttlinger**
## Things done

Packaged the ATL* logic reasoner TATL* (https://atila.ibisc.univ-evry.fr/tableau_ATL_star/)

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ta/tatl/package.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+    patchelf \
+      --set-interpreter $(cat $NIX_CC/nix-support/dynamic-linker) \
+      $out/bin/tatl
+
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "The implementation of a tableau-based decision procedure for the full Alternating-time Temporal Logic (ATL*).";
```
Comment (FliegendeWurst) [inline pkgs/by-name/ta/tatl/package.nix:29]:
```suggestion
    description = "Implementation of a tableau-based decision procedure for the full Alternating-time Temporal Logic (ATL*)";
```

File: pkgs/by-name/ta/tatl/package.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    install -m755 tatl $out/bin/
+
+    patchelf \
+      --set-interpreter $(cat $NIX_CC/nix-support/dynamic-linker) \
+      $out/bin/tatl
+
+    runHook postInstall
+  '';
+
+  meta = with lib; {
```
Comment (FliegendeWurst) [inline pkgs/by-name/ta/tatl/package.nix:28]:
```suggestion
  meta = {
```

File: pkgs/by-name/ta/tatl/package.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    description = "The implementation of a tableau-based decision procedure for the full Alternating-time Temporal Logic (ATL*).";
+    homepage = "https://atila.ibisc.univ-evry.fr/tableau_ATL_star/";
+    license = licenses.unfree; # free download on their website. Given that this is a reasearch tool and not a commercial product the intent clearly is to allow use and they simply did not specify a license.
+    maintainers = with maintainers; [ mgttlinger ];
```
Comment (FliegendeWurst) [inline pkgs/by-name/ta/tatl/package.nix:32]:
```suggestion
    # free download on their website. Given that this is a reasearch tool and not a commercial product the intent clearly is to allow use and they simply did not specify a license.
    license = lib.licenses.unfree;
    maintainers = with lib.maintainers; [ mgttlinger ];
```

File: pkgs/by-name/ta/tatl/package.nix [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  stdenv,
+  lib,
+  fetchzip,
+}:
+
+stdenv.mkDerivation {
+  name = "TATL*";
+  version = "20241001"; # They have no version number so using the time of writing this package as version.
+  src = fetchzip {
+    url = "https://atila.ibisc.univ-evry.fr/tableau_ATL_star/bin/tatl_star.tar.gz";
```
Comment (FliegendeWurst) [inline pkgs/by-name/ta/tatl/package.nix:11]: If they update the package without changing the URL, you need to use an archived copy. Unfortunately the wayback machine currently doesn't accept new archives.

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]

File: pkgs/by-name/ta/tatl/package.nix (hunk 4 above)
Comment (mgttlinger) [reply pkgs/by-name/ta/tatl/package.nix:11]: It's also questionable whether an archive URL for a nonfree package would even be legal, given that you would be distributing the package.

Comment (FliegendeWurst) [reply pkgs/by-name/ta/tatl/package.nix:11]: I suppose we could ask the developers whether they would be willing to version their archives.

Comment (mgttlinger) [reply pkgs/by-name/ta/tatl/package.nix:11]: Ok. I'll see if I can track down the developers, ask them, and possibly also license the tool while they're at it.

Comment (FliegendeWurst) [reply pkgs/by-name/ta/tatl/package.nix:11]: Did you get a response?

Comment (mgttlinger) [reply pkgs/by-name/ta/tatl/package.nix:11]: More or less. I'm in contact with the author and we are talking about open sourcing TATL which would make this whole package much nicer.

**Review by FliegendeWurst — APPROVED** [review]

**Comment by mgttlinger** [comment]
@FliegendeWurst Thanks for pushing me. Now, the tool is open source, which it wouldn't have been otherwise!

**Comment by mgttlinger** [comment]
Not sure what to make of the ofborg failure. Can someone enlighten me what is wrong here?

**Comment by FliegendeWurst** [comment]
That was due to another PR. You can force push here without changes to trigger ofborg again

**Comment by mgttlinger** [comment]
@ofborg eval

**Comment by FliegendeWurst** [comment]
@ofborg build tatl

**Comment by mgttlinger** [comment]
Ok, looks good now.

**Pull request merged**